### DISCUSSION TITLE
Upload Data directory should allow for customizing the directory via environment variable

### DISCUSSION THREAD

**Discussion by acherla** [StreetLamb/tribe#126] [Ideas]
Currently the app upload directory base path is statically defined via the following basepath /app/upload-data/ in uploads.py.
Instead allow for this to be an optionally configured directory that can be defined as an environment variable in order to modify this on the fly.  This would allow a user to specify their own shared PVC mount directory and specify a different path as needed.  
IE: APP_UPLOAD_DIRECTORY=/app/upload-data/

**Comment by StreetLamb** [comment]
Hi @acherla, thanks for the suggestion! However, I'm curious about the use cases where users would want to customise the directory?

**Reply by acherla** [reply]
Just a general configuration id argue that should be set by the user.  For example if i need to mount the docker image to a PVC due to security constraints it may require defining a different base path for uploads.  Otherwise itll default to using the docker image mount volume which id argue wouldnt be ideal.
